src/main/java/com/zufar/onlinestore/user/validation/UserDataValidator.java

```diff
@@ -0,0 +1,56 @@
+package com.zufar.onlinestore.user.validation;
+
+import com.zufar.onlinestore.user.entity.UserEntity;
+import com.zufar.onlinestore.user.exception.UserAlreadyRegisteredException;
+import com.zufar.onlinestore.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Component
+public class UserDataValidator {
```

Comment:
You should create A Custom Validator Annotation with Spring like that:
https://reflectoring.io/bean-validation-with-spring-boot/

Reply:
Done